platforms/hyperledger-fabric/configuration/roles/create/crypto/orderer/tasks/orderer.yaml

```diff
@@ -186,14 +185,16 @@
     cat {{ orderer.name }}-{{ component_name }}.pem > {{ orderer.name }}-{{ component_name }}-certchain.pem
     cat ./ca/ca.{{ component_name }}-cert.pem >> {{ orderer.name }}-{{ component_name }}-certchain.pem
   when: get_orderer_secret.resources|length == 0 and vault_orderer_ambassador.failed == True
-
+  changed_when: false
 
 # This task creates the Ambassador TLS credentials for orderer
 - name: Create the Ambassador credentials
   shell: |
     KUBECONFIG={{ kubernetes.config_file }} kubectl create secret tls {{orderer.name}}-{{ component_name }}-ambassador-certs --cert=./build/crypto-config/ordererOrganizations/{{ component_name }}/{{orderer.name}}-{{ component_name }}-certchain.pem --key=./build/crypto-config/ordererOrganizations/{{ component_name }}/{{orderer.name}}-{{ component_name }}.key -n default
   when: get_orderer_secret.resources|length == 0
-
+  changed_when: false
```

Comment:
why notest?

Reply:
This actually was placed because it was failing in converge check in the case where we assumed that the certificates were already present. So while creating Ambassador credentials, it was only taking into consideration whether only ambassador credentials exists or not and NOT whether the orderer ambassador secrets are created. So This was running okay when vault is empty from start but not when vault is pre filled. So, then i added another condition check at 'Create the Ambassador credentials' ie vault_orderer_ambassador.failed == True.
It then worked. both for fresh vault and pre vault setup scenario.
Also I checked the fabric Deployment with the current changes. Works fine